OpenQA/Isotovideo/Utils.pm

```diff
@@ -75,10 +75,14 @@ sub _clone_bare_repo ($clone_url, $clone_depth, $clone_cmd, $cache_dir, $handle_
     $handle_output->($?, qx{$clone_cmd --bare --depth='$clone_depth' '$clone_url' '$cache_dir' 2>&1});
 }
 
-sub _fetch_new_refs ($clone_url, $cache_dir, $branch_args, $handle_output) {
+sub _fetch_new_refs ($clone_url, $cache_dir, $branch_arg, $handle_output) {
     bmwqemu::fctinfo "Updating Git cache for '$clone_url' under '$cache_dir'";
-    $handle_output->($?, qx{git -C "$cache_dir" fetch origin $branch_args 2>&1 2>&1});
-    $handle_output->($?, qx{git -C "$cache_dir" branch --force $branch_args FETCH_HEAD 2>&1 2>&1}) if $branch_args;
+    if ($branch_arg eq "") {
+        # get default branch (usually "master") from remote repo if $branch_arg is empty
+        $branch_arg = "'" . `git ls-remote --symref '$clone_url' HEAD | awk -F'[/\\t]' 'NR == 1 {print \$3}'` . "'";
```

Comment:
I guess this generally makes sense. The code still needs proper error handling, though. It would probably also make more sense to parse the output in Perl and also handle the error when the output is not what we expect.

```suggestion
        $handle_output->($?, my @remote_refs = `git ls-remote --symref '$clone_url' HEAD`);
        my $remote_ref = (split(' ', $remove_refs[-1]))[2];
        die "Unable to parse default branch of remote repo '$clone_url'.\n";
        $branch_arg = "'$remote_ref'";
```

(Probably the splitting needs to be adjusted. I just wrote something that *might* be the Perl equivalent to the `awk` call.)

(If you really want to keep using `akw` you probably need to add it to the apparmor profile of the openQA worker.)